File: 849-maximize-distance-to-closest-person/849-maximize-distance-to-closest-person.py
class Solution:
    def maxDistToClosest(self, seats: List[int]) -> int:
        left_index = -1
        size = len(seats)
        i = 0
        max_nearest = 1
        while i<size:
            if seats[i]==1:
                max_nearest = max(max_nearest, i-0 if left_index==-1 else (i-left_index)//2)
                left_index = i
            i += 1
        max_nearest = max(max_nearest, i-left_index-1)
        return max_nearest

File: 849-maximize-distance-to-closest-person/849-maximize-distance-to-closest-person.java
class Solution {
    public int maxDistToClosest(int[] seats) {
        int n = seats.length;
        int[] distances = new int[n];
        int maxDis = 0;
        int nearSeatIndex=-n;
        for(int i=0; i<n; i++){
            if(seats[i]==1){
                nearSeatIndex=i;
            }
            distances[i] = i-nearSeatIndex;
        }
        nearSeatIndex=2*n;
        for(int i=n-1; i>=0; i--){
            if(seats[i]==1){
                nearSeatIndex=i;
            }
            distances[i] = Math.min(distances[i], nearSeatIndex-i);
            maxDis = Math.max(maxDis, distances[i]);
        }
        return maxDis;
    }
}